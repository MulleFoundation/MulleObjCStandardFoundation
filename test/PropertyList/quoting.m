//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>


static void  code_decode( id value)
{
   NSData   *data;
   id       decoded;

   printf( "%s->",
      [[value description] UTF8String]);
   fflush( stdout);

   @try
   {
      data   = [NSPropertyListSerialization dataFromPropertyList:value
                                                          format:NSPropertyListOpenStepFormat
                                                errorDescription:NULL];
      decoded = [NSPropertyListSerialization propertyListFromData:data
                                                 mutabilityOption:0
                                                           format:NULL
                                                 errorDescription:NULL];

      if( ! [[value description] isEqualToString:[decoded description]])
         printf( "*MISMATCH*");

      printf( "%s\n",
         [[decoded description] UTF8String]);
   }
   @catch( NSException *exception)
   {
      printf( "exception: %s\n", [[exception reason] UTF8String]);
   }
}



int main(int argc, const char * argv[])
{
   int   i;

   for( i = 7; i <= 13; i++)
      code_decode( @{ [NSNumber numberWithDouble:(double) i / 10]: [NSString stringWithFormat:@"%c", i] });
   for( i = 27; i <= 27; i++)
      code_decode( @{ [NSNumber numberWithDouble:(double) i / 10]: [NSString stringWithFormat:@"%c", i] });
   for( i = 32; i <= 126; i++)
      code_decode( @{ [NSNumber numberWithDouble:(double) i / 10]: [NSString stringWithFormat:@"%c", i] });
   return( 0);
}
