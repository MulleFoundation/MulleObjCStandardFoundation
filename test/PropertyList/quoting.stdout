{
   "0.7" = "\a";
}->{
   "0.7" = "\a";
}
{
   "0.8" = "\b";
}->{
   "0.8" = "\b";
}
{
   "0.9" = "\t";
}->{
   "0.9" = "\t";
}
{
   1 = "\n";
}->{
   1 = "\n";
}
{
   "1.1" = "\v";
}->{
   "1.1" = "\v";
}
{
   "1.2" = "\f";
}->{
   "1.2" = "\f";
}
{
   "1.3" = "\r";
}->{
   "1.3" = "\r";
}
{
   "2.7" = "\e";
}->{
   "2.7" = "\e";
}
{
   "3.2" = " ";
}->{
   "3.2" = " ";
}
{
   "3.3" = "!";
}->{
   "3.3" = "!";
}
{
   "3.4" = "\"";
}->{
   "3.4" = "\"";
}
{
   "3.5" = "#";
}->{
   "3.5" = "#";
}
{
   "3.6" = "$";
}->{
   "3.6" = "$";
}
{
   "3.7" = "%";
}->{
   "3.7" = "%";
}
{
   "3.8" = "&";
}->{
   "3.8" = "&";
}
{
   "3.9" = "'";
}->{
   "3.9" = "'";
}
{
   4 = "(";
}->{
   4 = "(";
}
{
   "4.1" = ")";
}->{
   "4.1" = ")";
}
{
   "4.2" = "*";
}->{
   "4.2" = "*";
}
{
   "4.3" = "+";
}->{
   "4.3" = "+";
}
{
   "4.4" = ",";
}->{
   "4.4" = ",";
}
{
   "4.5" = "-";
}->{
   "4.5" = "-";
}
{
   "4.6" = ".";
}->{
   "4.6" = ".";
}
{
   "4.7" = "/";
}->{
   "4.7" = "/";
}
{
   "4.8" = 0;
}->{
   "4.8" = 0;
}
{
   "4.9" = 1;
}->{
   "4.9" = 1;
}
{
   5 = 2;
}->{
   5 = 2;
}
{
   "5.1" = 3;
}->{
   "5.1" = 3;
}
{
   "5.2" = 4;
}->{
   "5.2" = 4;
}
{
   "5.3" = 5;
}->{
   "5.3" = 5;
}
{
   "5.4" = 6;
}->{
   "5.4" = 6;
}
{
   "5.5" = 7;
}->{
   "5.5" = 7;
}
{
   "5.6" = 8;
}->{
   "5.6" = 8;
}
{
   "5.7" = 9;
}->{
   "5.7" = 9;
}
{
   "5.8" = ":";
}->{
   "5.8" = ":";
}
{
   "5.9" = ";";
}->{
   "5.9" = ";";
}
{
   6 = "<";
}->{
   6 = "<";
}
{
   "6.1" = "=";
}->{
   "6.1" = "=";
}
{
   "6.2" = ">";
}->{
   "6.2" = ">";
}
{
   "6.3" = "\?";
}->{
   "6.3" = "\?";
}
{
   "6.4" = "@";
}->{
   "6.4" = "@";
}
{
   "6.5" = A;
}->{
   "6.5" = A;
}
{
   "6.6" = B;
}->{
   "6.6" = B;
}
{
   "6.7" = C;
}->{
   "6.7" = C;
}
{
   "6.8" = D;
}->{
   "6.8" = D;
}
{
   "6.9" = E;
}->{
   "6.9" = E;
}
{
   7 = F;
}->{
   7 = F;
}
{
   "7.1" = G;
}->{
   "7.1" = G;
}
{
   "7.2" = H;
}->{
   "7.2" = H;
}
{
   "7.3" = I;
}->{
   "7.3" = I;
}
{
   "7.4" = J;
}->{
   "7.4" = J;
}
{
   "7.5" = K;
}->{
   "7.5" = K;
}
{
   "7.6" = L;
}->{
   "7.6" = L;
}
{
   "7.7" = M;
}->{
   "7.7" = M;
}
{
   "7.8" = N;
}->{
   "7.8" = N;
}
{
   "7.9" = O;
}->{
   "7.9" = O;
}
{
   8 = P;
}->{
   8 = P;
}
{
   "8.1" = Q;
}->{
   "8.1" = Q;
}
{
   "8.2" = R;
}->{
   "8.2" = R;
}
{
   "8.3" = S;
}->{
   "8.3" = S;
}
{
   "8.4" = T;
}->{
   "8.4" = T;
}
{
   "8.5" = U;
}->{
   "8.5" = U;
}
{
   "8.6" = V;
}->{
   "8.6" = V;
}
{
   "8.7" = W;
}->{
   "8.7" = W;
}
{
   "8.8" = X;
}->{
   "8.8" = X;
}
{
   "8.9" = Y;
}->{
   "8.9" = Y;
}
{
   9 = Z;
}->{
   9 = Z;
}
{
   "9.1" = "[";
}->{
   "9.1" = "[";
}
{
   "9.2" = "\\";
}->{
   "9.2" = "\\";
}
{
   "9.3" = "]";
}->{
   "9.3" = "]";
}
{
   "9.4" = "^";
}->{
   "9.4" = "^";
}
{
   "9.5" = "_";
}->{
   "9.5" = "_";
}
{
   "9.6" = "`";
}->{
   "9.6" = "`";
}
{
   "9.7" = a;
}->{
   "9.7" = a;
}
{
   "9.8" = b;
}->{
   "9.8" = b;
}
{
   "9.9" = c;
}->{
   "9.9" = c;
}
{
   10 = d;
}->{
   10 = d;
}
{
   "10.1" = e;
}->{
   "10.1" = e;
}
{
   "10.2" = f;
}->{
   "10.2" = f;
}
{
   "10.3" = g;
}->{
   "10.3" = g;
}
{
   "10.4" = h;
}->{
   "10.4" = h;
}
{
   "10.5" = i;
}->{
   "10.5" = i;
}
{
   "10.6" = j;
}->{
   "10.6" = j;
}
{
   "10.7" = k;
}->{
   "10.7" = k;
}
{
   "10.8" = l;
}->{
   "10.8" = l;
}
{
   "10.9" = m;
}->{
   "10.9" = m;
}
{
   11 = n;
}->{
   11 = n;
}
{
   "11.1" = o;
}->{
   "11.1" = o;
}
{
   "11.2" = p;
}->{
   "11.2" = p;
}
{
   "11.3" = q;
}->{
   "11.3" = q;
}
{
   "11.4" = r;
}->{
   "11.4" = r;
}
{
   "11.5" = s;
}->{
   "11.5" = s;
}
{
   "11.6" = t;
}->{
   "11.6" = t;
}
{
   "11.7" = u;
}->{
   "11.7" = u;
}
{
   "11.8" = v;
}->{
   "11.8" = v;
}
{
   "11.9" = w;
}->{
   "11.9" = w;
}
{
   12 = x;
}->{
   12 = x;
}
{
   "12.1" = y;
}->{
   "12.1" = y;
}
{
   "12.2" = z;
}->{
   "12.2" = z;
}
{
   "12.3" = "{";
}->{
   "12.3" = "{";
}
{
   "12.4" = "|";
}->{
   "12.4" = "|";
}
{
   "12.5" = "}";
}->{
   "12.5" = "}";
}
{
   "12.6" = "~";
}->{
   "12.6" = "~";
}
